.github/workflows/node-windows.yml

```diff
@@ -17,21 +17,18 @@ jobs:
 
     strategy:
       matrix:
-        node: ['18', '16', '14']
+        node: ['20', '18', '16']
 
     name: Node v${{ matrix.node }}
     steps:
       - name: Configure git line-breaks
         run: git config --global core.autocrlf false
 
       - name: Checkout Commit
-        uses: actions/checkout@v1
-
-      - name: Checkout Master
-        run: git branch -f master origin/master
+        uses: actions/checkout@v4
```

Comment:
that's the reasoning here? opening PRs on open source projects means explaining changes so that maintainers and onlookers understand your changes.

Reply:
And maintainers of open source project should avoid to push on master partially changes breaking the CI :-) 

The checkout forcing master branch usage was unneeded since the GH actions has been fixed to properly use the default repo branch, even with the v1 major version.

Reply:
An CI run approval is needed to ensure some of the changes are working as intended.

Reply:
This might be language barrier but I'm not digging the words you're using; they are condescending.

> The checkout forcing master branch usage was unneeded since the GH actions has been fixed to properly use the default repo branch, even with the v1 major version.

Please cite your source.

Reply:
> This might be language barrier but I'm not digging the words you're using; they are condescending.

It's just a fact: code partially tested has be been pushed directly on master breaking the CI. It's neither a good practice nor recommended. Facts pointing has nothing to do with condescending. 

> > The checkout forcing master branch usage was unneeded since the GH actions has been fixed to properly use the default repo branch, even with the v1 major version.
> 
> Please cite your source.

The force-depth tunable to checkout branches is here since the very first release. 

The latest code in that PR is working properly on the forked repo.

Reply:
I'm really not digging the vibe of the discourse here. I asked for a cited source, not additional hyperbole. If you'd like to continue the PR, please provide a link to a commit or issue in the `actions/checkout` repo that we can reference. And I kindly request that your replies not continue to be combative. If that isn't acceptable please consider closing your PR. We don't expect everyone to be chill, but with the little time maintainers have, we're not going to engage with contributors who are not.

Reply:
> I'm really not digging the vibe of the discourse here. I asked for a cited source, not additional hyperbole. If you'd like to continue the PR, please provide a link to a commit or issue in the `actions/checkout` repo that we can reference. 

The force-depth tunable to checkout branches is here since probably minor revision of the version 1 or the version 2. The reference to its introduction has no importance, the rationale on its usage is explained in the description. 

+ The PR is working as intended
+ The changes are detailed
+ The regression on PRs validation is fixed
+ The project codeflow using PRs will be working again  

That's all that PR is meant for.